u4-databases/app.js

```diff
@@ -0,0 +1,36 @@
+const express = require('express')
+const expHbs = require('express-handlebars')
+const { json, urlencoded } = require('body-parser')
+const { bookDetails, bookSearch, bookCreateForm, bookCreate } = require('./features/book-controller.js')
+const { iterationList, iterationDetails, studentDetails, courseList, courseDetails } = require('./features/uni-controller.js')
+
+const app = express()
+
+// Templates
+app.set('views', './views')
+app.set('view engine', 'hbs')
+app.engine('hbs', expHbs({
+  extname: '.hbs',
+  defaultLayout: false,
+  partialsDir: ['./views/partials', './views/layouts']
+}))
+
+// Middleware
+app.use(express.static('public'))
+app.use(json())
+app.use(urlencoded({ extended: false }))
+
+// Routes
+app.get('/books', bookSearch)
+app.post('/books', bookCreate)
+app.get('/books/new', bookCreateForm)
+app.get('/books/:isbn13', bookDetails)
+app.get('/images/:catchall', (_req, res) => res.redirect('/images/404.jpg'))
```

Comment:
ฉันจะใส่ "image catch-all" เป็นเส้นทางสุดท้ายของคุณ
I would put the "image catch-all" as your last route.